functional_tests/test_db_online_offline_resync.py

```diff
@@ -0,0 +1,441 @@
+    assert(resync_result['payload']['changes'] == num_docs * num_users)
+    assert(resync_result['status_code'] == 200)
+
+    time.sleep(5)
+    global_cache = list()
+    for user in user_objects:
+        global_cache.append(user.cache)
+
+    all_docs = {k: v for user_cache in global_cache for k, v in user_cache.items()}
+
+    verify_changes(user_x, expected_num_docs=expected_docs, expected_num_revisions=num_revisions, expected_docs=all_docs)
+
+    end = time.time()
+    log.info("Test ended.")
+    log.info("Main test duration: {}".format(end - init_completed))
+    log.info("Test setup time: {}".format(init_completed - start))
+    log.info("Total Time taken: {}s".format(end - start))
+
+
+# implements scenario: 12 and 13
+# While DB is running _resync make REST API calls against DB for supported offline operations,
+# GET db status, PUT DB config, expected result calls should return status 200.
+# #13
+# With DB running a _resync, make REST API call to get DB runtime details /db/,
+# expected result 'state' property with value 'Resyncing' is returned.
+@pytest.mark.sanity
+@pytest.mark.dbonlineoffline
+@pytest.mark.parametrize("num_users", [5])
+@pytest.mark.parametrize("num_docs", [100])
+@pytest.mark.parametrize("num_revisions", [5])
+def test_bucket_online_offline_resync_with_offline(cluster, num_users, num_docs, num_revisions):
+    log.info("Starting test...")
+    start = time.time()
+
+    cluster.reset("bucket_online_offline/db_online_offline_access_all.json")
+
+    init_completed = time.time()
+    log.info("Initialization completed. Time taken:{}s".format(init_completed - start))
+
+    users = ["User-" + str(i) for i in range(num_users)]
+    num_channels = 1
+    channels = ["channel-" + str(i) for i in range(num_channels)]
+    password = "password"
+    user_objects = []
+    use_uuid_names = True
+
+    sgs = cluster.sync_gateways
+
+    admin = Admin(sgs[0])
+
+    # Register User
+    log.info("Register User")
+    user_objects = admin.register_bulk_users(target=sgs[1], db="db", name_prefix="User",
+                                             number=num_users, password=password, channels=channels)
+    user_x = admin.register_user(target=sgs[0], db="db", name="User-X", password="password", channels=["channel_x"])
+
+    # Add User
+    log.info("Add docs")
+    in_parallel(user_objects, 'add_docs', num_docs)
+
+    # Update docs
+    log.info("Update docs")
+    in_parallel(user_objects, 'update_docs', num_revisions)
+
+    time.sleep(10)
+
+    # Get changes for all users
+    in_parallel(user_objects, 'get_changes')
+
+    # every user should have same number of docs
+    # total/expected docs = num_users * num_docs
+    recieved_docs = in_parallel(user_objects, 'get_num_docs')
+
+    expected_docs = num_users * num_docs
+    for user_obj, docs in recieved_docs.items():
+        log.info('User {} got {} docs, expected docs: {}'.format(user_obj.name, docs, expected_docs))
+        assert docs == 1 + expected_docs
```

Comment:
Can you update get_num_docs to work with both original sync_gateway mode as well as distributed index mode. (i.e. ignore _user docs)